reports/report_generation.go

```diff
@@ -27,158 +26,48 @@ import (
+func GenerateReport(ctx context.Context, aa aws.AwsAccount, aas []aws.AwsAccount, date time.Time) (errs map[string]error) {
 	logger := jsonlog.LoggerFromContextOrDefault(ctx)
-	reportDate := formatDate(date)
-	logger.Info("Generating spreadsheet for account", map[string]interface{}{
-		"account": aa,
-		"date":    reportDate,
-	})
-	errs = make(map[string]error)
-
-	if tx, err := db.Db.BeginTx(ctx, nil); err == nil {
-		sheets, errs := getSpreadsheetData(ctx, aa, date, tx)
-		if len(errs) > 0 {
-			logger.Error("Error while getting spreadsheet data", errs)
-		}
-
-		file, spreadsheetErrors := generateSpreadsheet(ctx, aa, reportDate, sheets)
-		if len(spreadsheetErrors) > 0 {
-			logger.Error("Error while generating spreadsheet", spreadsheetErrors)
-			for errorKey := range spreadsheetErrors {
-				if _, ok := errs[errorKey]; !ok {
-					errs[errorKey] = spreadsheetErrors[errorKey]
-				}
-			}
-		}
-		errs["speadsheetError"] = saveSpreadsheet(ctx, file, false)
+	now := time.Now()
+	var masterReport bool
+	var reportDate string
+	if date.IsZero() {
+		reportDate = fmt.Sprintf("%s%s", (now.Month() - 1).String(), strconv.Itoa(now.Year()))
 	} else {
-		errs["speadsheetError"] = err
+		reportDate = fmt.Sprintf("%s%s", (date.Month()).String(), strconv.Itoa(date.Year()))
+	}
+	if aas == nil {
+		aas = []aws.AwsAccount{aa}
+		masterReport = false
+		logger.Info("Generating spreadsheet for account", map[string]interface{}{
+			"account": aa,
+			"date":    reportDate,
+		})
+	} else {
+		masterReport = true
+		logger.Info("Generating spreadsheet for accounts", map[string]interface{}{
+			"masterAccount": aa,
+			"accounts":      aas,
+			"date":          reportDate,
+		})
 	}
-	return
-}
-
-func GenerateMasterReport(ctx context.Context, aa aws.AwsAccount, aas []aws.AwsAccount, date time.Time) (errs map[string]error) {
-	logger := jsonlog.LoggerFromContextOrDefault(ctx)
-	reportDate := formatDate(date)
-	logger.Info("Generating master spreadsheet for accounts", map[string]interface{}{
-		"masterAccount": aa,
-		"accounts":      aas,
-		"date":          reportDate,
-	})
 	errs = make(map[string]error)
+	file := createSpreadsheet(aa, reportDate)
 
 	if tx, err := db.Db.BeginTx(ctx, nil); err == nil {
-		sheets, errs := getMasterSpreadsheetData(ctx, aas, date, tx)
-		if len(errs) > 0 {
-			logger.Error("Error while getting spreadsheet data", errs)
-		}
-
-		file, spreadsheetErrors := generateSpreadsheet(ctx, aa, reportDate, sheets)
-		if len(spreadsheetErrors) > 0 {
-			logger.Error("Error while generating spreadsheet", spreadsheetErrors)
-			for errorKey := range spreadsheetErrors {
-				if _, ok := errs[errorKey]; !ok {
-					errs[errorKey] = spreadsheetErrors[errorKey]
-				}
+		for _, module := range modules {
+			err := module.GenerateSheet(ctx, aas, date, tx, file.File)
+			if err != nil {
+				errs[module.ErrorName] = err
 			}
 		}
-		errs["speadsheetError"] = saveSpreadsheet(ctx, file, true)
+
+		/* Remove the first sheet since it is unused */
```

Comment:
For one-lilne comments `//` needs to be used 

Also, `/* ... */` comments should only be used while debugging, block code comments for documentation should be like this : 
```
// this is a good
// comment block
```